modules/lambda/main.tf

```diff
@@ -1,9 +1,10 @@
 locals {
-  enabled             = module.this.enabled
-  iam_policy_enabled  = local.enabled && (try(length(var.iam_policy), 0) > 0 || var.policy_json != null)
-  s3_bucket_full_name = var.s3_bucket_name != null ? format("%s-%s-%s-%s-%s", module.this.namespace, module.this.tenant, module.this.environment, module.this.stage, var.s3_bucket_name) : null
```

Comment:
this removes support for dynamically selecting a bucket name. For example

```
components:
  terraform:
     s3-bucket:
        vars:
          name: foo # --> ends up with "%s-%s-%s-%s-%s" name

     lambda:
        vars:
           s3_bucket_name: foo # --> find the "%s-%s-%s-%s-%s" name
```

previously we could dynamically build the rest of the bucket name for any stage

Reply:
I dont know if anyone actually uses it that way. Alternatively, we could have a different variable such as `var. s3_bucket_full_name` if you wanted to specify the literal name

Reply:
I've re-added bucket_name calculation. I didn't think that the `s3_bucket_name` with calculation was all that helpful. admittedly I forgot about the case where you might want to declare the bucket name once because you actually do have a bucket in each plat account. then we want formatting so you don't need an override per account